### PR TITLE
feat: add oauth api for gitlab

### PR DESCRIPTION
**Pull request by JiekunZhang**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes 

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer
@wzqnls @supereagle  @zhujian7 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhujian7** [comment]
The Travis CI build failed, please fix it.

File: pkg/scm/provider/helper.go [hunk 1]
```diff
@@ -16,7 +16,14 @@ limitations under the License.
 
 package provider
 
-import "strings"
+import (
+	"strings"
+	"fmt"
+	"golang.org/x/oauth2"
+	"github.com/caicloud/cyclone/pkg/api"
```
Comment (supereagle) [inline pkg/scm/provider/helper.go]: please format

File: pkg/server/router/oauth.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	if err != nil {
+		httputil.ResponseWithError(response,err)
+	}
+
+	redirectURL, err := provider.Authcallback(code, state)
+	if err != nil {
+		httputil.ResponseWithError(response, err)
+		return
+	}
+	fmt.Println(redirectURL)
```
Comment (supereagle) [inline pkg/server/router/oauth.go]: use log instead of fmt.print

File: pkg/scm/provider/helper.go [hunk 3]
```diff
@@ -25,3 +32,47 @@ func parseURL(url string) (string, string) {
...
+// getConfig pack the information into oauth.config that is used to get token
+// ClientID、ClientSecret, these values are used to assemble the token request url
+// and they come from github or gitlab or others by registering application information.
+func getConfig(scmType string) (*oauth2.Config, error) {
+	var clientID, clientSecret, redirectURL, authURL, tokenURL string
+	var scopes []string
+
+	switch scmType {
+
+	case api.GITLAB:
```
Comment (supereagle) [inline pkg/scm/provider/helper.go]: use api.Gitlab

File: pkg/server/router/oauth.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	scmType := request.PathParameter("type")
+	provider, err := scm.GetSCMProvider(api.Gitlab)
+	if err != nil {
+		httputil.ResponseWithError(response,err)
+	}
+	url, err := provider.GetAuthCodeURL(state, scmType)
+	if err != nil {
+		httputil.ResponseWithError(response,err)
+	}
+	fmt.Printf("code url: %s",url)
```
Comment (supereagle) [inline pkg/server/router/oauth.go]: use log instead of fmt.print

File: pkg/scm/provider/svn.go [hunk 5]
```diff
@@ -110,3 +110,11 @@ func (s *SVN) CreateWebHook(scm *api.SCMConfig, repoURL string, webHook *scm.Web
 func (s *SVN) DeleteWebHook(scm *api.SCMConfig, repoURL string, webHookUrl string) error {
 	return nil
 }
+
+// TODO: oauth by SVN
+func (s *SVN) GetAuthCodeURL(state string, scmType string) (string, error) {
+	return "", nil
```
Comment (supereagle) [inline pkg/scm/provider/svn.go]: This function is not implemented, should return error: `return "", fmt.Errorf("Not implemented")`

Other not implemented functions also need to be fixed.

**Pull request merged**